src/metax_api/api/rest/base/serializers/contract_serializer.py

```diff
@@ -24,4 +26,12 @@ class Meta:
 
     def validate_contract_json(self, value):
         validate_json(value, self.context['view'].json_schema)
+        if self._operation_is_create:
+            self._validate_identifier_uniqueness(value)
         return value
+
+    def _validate_identifier_uniqueness(self, contract_json):
+        if Contract.objects.filter(contract_json__identifier=contract_json['identifier']).exists():
+            raise ValidationError({'identifier':
+                ['identifier %s already exists' % contract_json['identifier']]
+            })
```

Comment:
`[f"identifier {contract_json['identifier']} already exists"]` could be more clear